projects/client/src/app/pages/categories/pages/add/hooks/useAddCategory.ts

```diff
@@ -1,5 +1,47 @@
-const useAddCategory = () => {
-  const addCategory = () => {};
+import { toast } from 'sonner';
+
+import useNavigate from '@/app/hooks/useNavigate';
+import { CreateCategoryDto } from '@/models/category.model';
+import { createCategory } from '@/services/category.service';
+
+type Props = {
+  setIsNameCorrect: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsImageCorrect: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const useAddCategory = ({ setIsNameCorrect, setIsImageCorrect }: Props) => {
+  const { navigate } = useNavigate();
+
+  const addCategory = (category: CreateCategoryDto) => {
+    const response = toast.promise(createCategory(category), {
+      loading: 'Creating category...',
+      success: (res) => {
+        setIsNameCorrect(true);
+        setIsImageCorrect(true);
+        navigate('/admin/categories');
+        return `Category ${res.name} created successfully`;
+      },
+      error: (err) => {
+        const { message } = err.response.data;
+        let error = '';
```

Comment:
You have used const consistently which is good practice...  however, you used 'let' in this part. You could use const for destructuring the error message as well, right?